### PR TITLE
Balder/compute summary cache automatically in backend

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR
Especially how HwInfo is propagated.

### CODE REVIEW

**Comment by baldersheim** [comment]
@geirst Now it is complete and reviewable.

File: searchcore/src/vespa/searchcore/proton/server/bootstrapconfig.cpp [hunk 1]
```diff
@@ -29,29 +29,29 @@ BootstrapConfig::BootstrapConfig(
...
-    return equals<DocumenttypesConfig>(_documenttypes.get(),
-                                       rhs._documenttypes.get()) &&
+    return equals<DocumenttypesConfig>(_documenttypes.get(), rhs._documenttypes.get()) &&
         _repo.get() == rhs._repo.get() &&
         equals<ProtonConfig>(_proton.get(), rhs._proton.get()) &&
         equals<FiledistributorrpcConfig>(_fileDistributorRpc.get(), rhs._fileDistributorRpc.get()) &&
         equals<BucketspacesConfig>(_bucketspaces.get(), rhs._bucketspaces.get()) &&
-        equals<TuneFileDocumentDB>(_tuneFileDocumentDB.get(),
-                                   rhs._tuneFileDocumentDB.get());
+        equals<TuneFileDocumentDB>(_tuneFileDocumentDB.get(), rhs._tuneFileDocumentDB.get());
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/bootstrapconfig.cpp]: HwInfo should probably be part of the equals function as well?

**Comment by geirst** [comment]
Overall looks good, please look at my comment first

**Comment by baldersheim** [comment]
@geirst equal implemented and tested

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**